internal/driver/driver_test.go

```diff
@@ -1023,6 +1041,96 @@ func TestSymbolzAfterMerge(t *testing.T) {
 	}
 }
 
+func TestHttpsInsecure(t *testing.T) {
+	baseVars := pprofVariables
+	pprofVariables = baseVars.makeCopy()
+	defer func() { pprofVariables = baseVars }()
+
+	tlsConfig := &tls.Config{Certificates: []tls.Certificate{selfSignedCert(t)}}
+	donec := make(chan struct{}, 1)
+
+	l, err := tls.Listen("tcp", "localhost:0", tlsConfig)
+	if err != nil {
+		t.Fatalf("net.Listen: got error %v, want no error", err)
+	}
+	defer func() { <-donec }()
+	defer l.Close()
+
+	go func(donec chan<- struct{}) {
+		defer func() { donec <- struct{}{} }()
+		if got, want := http.Serve(l, nil), "use of closed"; !strings.Contains(got.Error(), want) {
+			t.Fatalf("Serve got error %v, want %q", got, want)
+		}
+	}(donec)
+
+	go func() {
+		deadline := time.Now().Add(5 * time.Second)
+		for time.Now().Before(deadline) {
+			// Simulate a hotspot function.
+		}
+	}()
+
+	outputTempFile, err := ioutil.TempFile("", "profile_output")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer os.Remove(outputTempFile.Name())
+	defer outputTempFile.Close()
+
+	f := emptyFlags()
+	o := setDefaults(nil)
+	o.Flagset = &f
+
+	f.args = []string{"https+insecure://" + l.Addr().String() + "/debug/pprof/profile?seconds=5"}
+	addFlags(&f, []string{
+		"top",
+		"symbolize=remote",
+		"output=" + outputTempFile.Name(),
+	})
+
+	if err := PProf(o); err != nil {
+		t.Fatalf("PProf(%v): got error %v, want no error", o, err)
+	}
+
+	b, err := ioutil.ReadFile(outputTempFile.Name())
+	if err != nil {
+		t.Fatalf("ReadFile(%s) got error %v, want no error", outputTempFile.Name(), err)
+	}
+	if got, want := string(b), "TestHttpsInsecure"; !strings.Contains(got, want) {
```

Comment:
Which platform are you running on? I checked with 1.7 and 1.8 and the test passes for me.

```
@aalexand0:pprof$ ( export PATH=~/sw/go1.7/bin:$PATH && export GOROOT=~/sw/go1.7 && go test -v ./internal/driver/... ) 2>&1 | grep Insecure
=== RUN   TestHttpsInsecure
--- PASS: TestHttpsInsecure (5.03s)
@aalexand0:pprof$ ( export PATH=~/sw/go1.8/bin:$PATH && export GOROOT=~/sw/go1.8 && go test -v ./internal/driver/... ) 2>&1 | grep Insecure
=== RUN   TestHttpsInsecure
--- PASS: TestHttpsInsecure (5.03s)
```